eeg_eyetracking_parser/_eeg_preprocessing.py

```diff
@@ -42,57 +77,131 @@ def create_eog_channels(raw, eog_channels=['VEOGT', 'VEOGB', 'HEOGL', 'HEOGR']):
     """
     Set Montage for EEG channel positions
     Here we use standard_1020 and thus electrode names and positions are according to international 10-20 system
+
+    Parameters
+    ----------
+    raw: instance of raw EEG data
+        A raw EEG data
+    montage_name: string, optional
+        A montage that contains channel positions
+    plot: bool, optional
+        True if plotting is to be done. By default, no plotting
+
+    Returns
+    -------
+    raw: instance of raw EEG data
+        A raw EEG data with channel locations
     """
     montage = mne.channels.make_standard_montage(montage_name)
     if plot:
         montage.plot()
     if 'Digi' in raw.info['ch_names']:
         raw.set_channel_types({'Digi': 'stim'})
     raw.set_montage(montage, match_case=False)
-    return raw
 
 
 def notch_filter(raw, frequencies_remove=(50, 100, 150, 200)):
     """
     Removes power line noise generated by EEG system
+
+    Parameters
+    ----------
+    raw: instance of raw EEG data
+        A raw EEG data
+    frequencies_remove: float, array of float, optional
+        Frequencies to remove from the data
+
+    Returns
+    -------
+    raw: instance of raw EEG data
+        A raw EEG data with certain frequencies that were filtered out
     """
     raw.notch_filter(frequencies_remove)
-    return raw
 
 
 def band_pass_filter(raw, lf=0.1, hf=40, plot=False):
     """
     High-pass and low-pass filter for EEG data (band-pass filter)
     High-pass removes slow drifts
     Low-pass removes the short-term fluctuations
+
+    Parameters
+    ----------
+    raw: instance of raw EEG data
+        A raw EEG data
+    lf: float, optional
+        Low-pass frequency edge
+    hf: float, optional
+        High-pass frequency edge
+    plot: bool, optional
+        True if plotting is to be done. By default, no plotting
+
+    Returns
+    -------
+    raw: instance of raw EEG data
+        A raw filtered EEG data
     """
-    raw_unfiltered = raw.copy()
+    if plot:
+        raw_unfiltered = raw.copy()
     raw.filter(l_freq=lf, h_freq=hf, picks='all')
     if plot:
         for title, data in zip(['Unfiltered', 'Bandpass filtered'], [raw_unfiltered, raw]):
             fig = data.plot_psd(fmax=250)
             fig.suptitle(title)
-    return raw
 
 
-def downsample_data(raw, srate=256):
+def downsample_data(raw, events, srate=256):
     """
     Downsample EEG data for computational efficacy
+
+    Parameters
+    ----------
+    raw: instance of raw EEG data
+        A raw EEG data
+    events: tuple
+        Event information as returned by `read_subject()`.
+    srate: float, optional
+        Sampling rate to convert to
+
+    Returns
+    -------
+    raw: instance of raw EEG data
+        A raw downsampled EEG data
+    events: tuple
+        Downsampled events.
     """
-    raw.resample(srate, npad="auto")
-    return raw
+    raw, events = raw.resample(srate, events=events, npad="auto")
```

Comment:
In this case, the raw and events objects are not modified, but rather new ones are generated. So here they need to be returned explicitly. (I'll do that, but just fyi.)